variables.tf

```diff
@@ -204,10 +205,48 @@ variable "requires_internet_access" {
   type        = bool
 }
 
-variable "with_appmesh" {
-  default     = false
-  description = "This services should be created with an appmesh proxy."
-  type        = bool
+variable "app_mesh_configuration" {
+  default = {
```

Comment:
the problem with using an object here: as soon as a client of this module changes one attribute of that object (like `enabled=true`), all attributes need to be set regardless if the defaults would be sufficient or not. 

Maybe using a `map(any)` would be more user friendly, although the (generated) documentation might not be as explicit as it it now.

Reply:
if tried explicitly typing this and reverted to the current state. Not sure what the best user experience would be. currently it would break when `app_mesh_configuration.enabled=false` but with routes defined.